### ISSUE TITLE
Deadlock

### ISSUE THREAD

**Issue by clsid2** [pinterf/xy-VSFilter#10]
There is a deadlock that sometimes occurs while unloading VSFilter. It happens when at same time SetupFRD function is called from inside the loop in CDirectVobSubFilter::ThreadProc().

Related commit, which only partially fixed it:
https://github.com/Cyberbeing/xy-VSFilter/commit/d7ccef214c78584424aab16c1684f8d307d25826

One solution would be to remove the lock inside the destructor CDirectVobSubFilter::~CDirectVobSubFilter()

To reproduce use MPC-HC with subtitle renderer set to VSFilter and a couple videos that don't have subtitles. Then quickly keep loading files (e.g. with PageUp/PageDown) and eventually it will hang.

**Comment by pinterf** [comment]
Just returned to the project, do you know whether this one was fixed already? Is it serious/really needs investigation? 
Yes, i know, the project is between dead/half-dead.

**Comment by clsid2** [comment]
Nope, not fixed afaik. It was pretty easy to trigger if I remember correctly.
